### PR TITLE
HTTP(S) Adapter Improvements and various query sanitizations

### PR DESCRIPTION
**Pull request by xq262144**
1. Support http/unprivileged mode, easy for testing and POC.
2. Support prepare statement and parameters in adapter query.
3. Support `application/json` content-type request in query.
4. Various query sanitization, including: sqlite special functions/internal tables/special queries.
5. Slow query logging support.
6. LevelDB storage page size adjustment.

### CODE REVIEW

File: xenomint/query_sanitizer.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+						return true, nil
+					})
+
+					return
+				}
+			}
+			return true, nil
+		}, walkNodes...)
+		if err != nil {
+			err = errors.Wrapf(err, "parse sql failed")
```
Comment (auxten) [inline xenomint/query_sanitizer.go]: missing args

File: xenomint/query_sanitizer.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+		switch stmt := statements[i].(type) {
+		case *sqlparser.Show:
+			origQuery = queryParts[i]
+
+			switch stmt.Type {
+			case "table":
+				if stmt.ShowCreate {
+					query = "SELECT sql FROM sqlite_master WHERE type = \"table\" AND tbl_name = \"" +
+						stmt.OnTable.Name.String() + "\" AND tbl_name NOT LIKE \"sqlite%\""
```
Comment (leventeliu) [inline xenomint/query_sanitizer.go]: Tooooo many backslashes, why not use back quote?

Comment (xq262144) [reply xenomint/query_sanitizer.go]: Good idea, working on that.

**Review by leventeliu — APPROVED** [review]
LGTM, approve.

**Review by auxten — APPROVED** [review]
LGTM

**Pull request merged**